gdrapi.c

```diff
@@ -335,40 +344,137 @@ static void gdr_init_cpu_flags()
 
 // note: more than one implementation may be compiled in
 
+static void unroll8_memcpy(void *map_d_ptr, const void *h_ptr, size_t size)
```

Comment:
Let me clarify the previous comment.

You define this unroll8_memcpy(map_d_ptr, h_ptr) function to copy from "h_ptr" to "map_d_ptr", right? However, you use it as unroll8_memcpy(h_ptr, map_d_ptr) in line 505. Functionally, this has no problem at all. However, you can see that the definitions (in term of human understanding, not the code) of "h_ptr" and "map_d_ptr" are still reversed here.

For example cudaMemcpy(dst, src), it tells exactly that which parameter is source and which is destination. Maintaining the code would be easier that way.

Reply:
agreed